File: src/sdk/python/examples/error_handling.py
from bridge_sdk import BridgeClient, BridgeError, BridgeValidationError
from common import REQUEST

try:
    result = BridgeClient.solver().route(REQUEST).result
    if not result["path_found"]:
        print("経路は見つかりませんでした。これは通信エラーではありません。")
except BridgeValidationError as exc:
    print(f"入力エラー: {exc}")
except BridgeError as exc:
    print(f"BRIDGE SDKエラー: {exc}")

File: src/sdk/typescript/examples/error-handling.ts
import { BridgeClient, BridgeError, BridgeValidationError } from "../src/index.js";
import { request } from "./common.js";
try {
  const result = (await (await BridgeClient.solver()).route(request)).result;
  if (!result.path_found) console.log("経路なしは通信エラーではありません");
} catch (error) {
  if (error instanceof BridgeValidationError) console.error(`入力エラー: ${error.message}`);
  else if (error instanceof BridgeError) console.error(`SDKエラー: ${error.message}`);
  else throw error;
}
